### DISCUSSION TITLE
Allow Extensions to display in the side panel

### DISCUSSION THREAD

**Discussion by gianpaj** [zed-industries/zed#19422] [Feature Requests]
To display UI in extensions like [Todo MD](https://marketplace.visualstudio.com/items?itemName=usernamehw.todo-md), [GitLens](https://marketplace.visualstudio.com/items?itemName=eamodio.gitlens), etc.

I don't know exactly how to toggle the Side panel of each Extension. Maybe like VSCode
<img width="300" alt="image" src="https://github.com/user-attachments/assets/f41a05a1-0b79-4b24-bf63-8a49296a57c1"> <img width="300" alt="image" src="https://github.com/user-attachments/assets/6fe5f4d9-12d1-4b27-8fae-a0a88ffc73da">

or from a dropdown of Extensions that have this functionality

tags: side panel, ui, drawer, interface

**Comment by 0x2CA** [comment]
I think the extended interface should not be realized in the next one or two years, the team is not ready to extend the interface is not ready, the future development is just pure logic extension, the current interface rendering is completely code implementation, chaining syntax call api, if you want to let the extension to use, not too safe, and also complex, the most ideal I think it should be the implementation of the SDL, using the type of xml descriptive language, the implementation of the interface description to implement the interface rendering, I think there is still a lot of work to be done on these.

https://www.reddit.com/r/ZedEditor/comments/1gpvff3/comment/lx46r4f/
https://www.reddit.com/r/ZedEditor/comments/1gpvff3/comment/lx4awla/

**Comment by JosephTLyons** [comment]
Thanks for reaching out. We want to better support our community, and the best way for us to do that is through our official Discord. The team is active there and happy to help.

We’re closing the “Help and General Q&A” category on GitHub Discussions and directing all support questions to Discord going forward.

https://zed.dev/support

**Reply by gianpaj** [reply]
Hi @JosephTLyons can this be converted into a GH discussion please? I feel Discord things can get lost very quickly – especially Feature requests like this
